dateparser/search_dates/search.py

```diff
@@ -0,0 +1,278 @@
+import re
+from typing import List, Dict
+from string import punctuation
+
+from dateparser.conf import apply_settings, check_settings, Settings
+from dateparser.date import DateDataParser
+from dateparser.search_dates.languages import SearchLanguages
+
+_drop_words = {"on", "of", "The"}  # cause annoying false positives
+_bad_date_re = re.compile(
+    # whole dates we black-list (can still be parts of valid dates)
+    "^("
+    + "|".join(
+        [
+            r"\d{1,3}",  # less than 4 digits
+            r"#\d+",  # this is a sequence number
+            # some common false positives below
+            r"[-/.]+",  # bare separators parsed as current date
+            r"\w\.?",  # one letter (with optional dot)
+            "an",
+        ]
+    )
+    + ")$"
+)
+
+_secondary_splitters = [
+    ",",
+    "،",
+    "——",
+    "—",
+    "–",
+    ".",
+]  # are used if no date object is found
+_punctuations = list(set(punctuation))
+
+
+def _get_relative_base(already_parsed):
+    if already_parsed:
+        return already_parsed[-1][1]
+    return None
+
+
+def _create_splits(text):
+    splited_objects = text.split()
+    splited_objects = [p for p in splited_objects if p and p not in _drop_words]
```

Comment:
I see that you changed the `_drop_words` from uppercase to lowercase. Are you sure we don't need to do:

`... and p.lower() not in _drop_words]`

?

Reply:
Hi, Marc, actually causes the above issue and prevents returning the correct date string. I am looking into replacing this approach altogether

Reply:
I have changed the filtering. Thanks to it new search_dates is now faster than the old implementation.

@noviluni